tests/integration/telemetry/stackdriver/stackdriver_filter_test.go

```diff
@@ -96,28 +98,41 @@ func unmarshalFromTemplateFile(file string, out proto.Message, clName string) er
 func TestStackdriverMonitoring(t *testing.T) {
 	framework.NewTest(t).
 		Run(func(ctx framework.TestContext) {
-			sendTraffic(t)
-
-			for _, cl := range ctx.Clusters() {
-				scopes.Framework.Infof("Validating Telemetry for Cluster %v", cl)
-				retry.UntilSuccessOrFail(t, func() error {
-					clName := cl.Name()
-					//Validate cluster names in telemetry below once https://github.com/istio/istio/issues/28125 is fixed.
-					if err := validateMetrics(t, serverRequestCount, clientRequestCount, clName); err != nil {
-						return err
-					}
-					if err := validateLogs(t, serverLogEntry, clName); err != nil {
-						return err
-					}
-					if err := validateTraces(t); err != nil {
-						return err
-					}
-					if err := validateEdges(t, clName); err != nil {
+
+			g, _ := errgroup.WithContext(context.Background())
+			for _, cltInstance := range clt {
+				g.Go(func() error {
+					err := retry.UntilSuccess(func() error {
+						if err := sendTraffic(t, cltInstance); err != nil {
```

Comment:
Why did you move sending traffic inside the loop? Shouldn't we just send traffic once and then wait for telemetry to show it?

Reply:
The problem is if the traffic fails telemetry never shows up

Or in another case, it was applying config then sending requests immediately. if its too fast, the config never applied

Reply:
I changed this behavior everywhere because many tests apply yaml files within the tests and it takes time for even deletion of that yaml to take effect...